### PR TITLE
Clean up Exec

### PR DESCRIPTION
**Pull request by aldonunez**

### CODE REVIEW

File: CVSym/CVSTI/DataSource.h [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #pragma once
 
 #include "IDataSource.h"
+#include <memory>
```
Comment (rainers) [inline CVSym/CVSTI/DataSource.h]: What was the error here? It compiles for me in VS2017 without the include.

File: DebugEngine/Debuggees/test1/test1.vcxproj [hunk 2]
```diff
@@ -86,7 +86,7 @@
       <DebugInformationFormat>EditAndContinue</DebugInformationFormat>
     </ClCompile>
     <Link>
-      <GenerateDebugInformation>true</GenerateDebugInformation>
+      <GenerateDebugInformation>DebugFull</GenerateDebugInformation>
```
Comment (rainers) [inline DebugEngine/Debuggees/test1/test1.vcxproj]: This is not accepted in VS2013, but the VS2017 compiled Concord plugin crashes when used inside VS2013. So I'm currently still building the release with VS2013 (that does not include the tests, though).

File: CVSym/CVSTI/DataSource.h (hunk 1 above)
Comment (aldonunez) [reply CVSym/CVSTI/DataSource.h]: It's complaining about not knowing what auto_ptr is.

Comment (aldonunez) [reply CVSym/CVSTI/DataSource.h]: I figured out what happened.

Before VS 2017, auto_ptr was defined in xmemory, which is included in memory. But, it's also included in vector and list, which are included in Common.h. That's why the compilers didn't use to complain. Beginning in VS 2017, auto_ptr is defined in memory.

Comment (aldonunez) [reply CVSym/CVSTI/DataSource.h]: But the question is, why does it compile for you in VS 2017?

File: DebugEngine/Debuggees/test1/test1.vcxproj (hunk 2 above)
Comment (aldonunez) [reply DebugEngine/Debuggees/test1/test1.vcxproj]: I'll change:
`<GenerateDebugInformation>DebugFull</GenerateDebugInformation>`
to
`<GenerateDebugInformation Condition="'$(PlatformToolset)' == 'v120' or  '$(PlatformToolset)' == 'v140'">true</GenerateDebugInformation>`
`<GenerateDebugInformation Condition="'$(PlatformToolset)' != 'v120' and '$(PlatformToolset)' != 'v140'">DebugFull</GenerateDebugInformation>`

I found that this works in VS 2015 and VS 2017. I don't have VS 2013.

File: CVSym/CVSTI/DataSource.h (hunk 1 above)
Comment (rainers) [reply CVSym/CVSTI/DataSource.h]: Probably not worth investigating further, it doesn't do much harm if added.

File: DebugEngine/Debuggees/test1/test1.vcxproj (hunk 2 above)
Comment (rainers) [reply DebugEngine/Debuggees/test1/test1.vcxproj]: Looks good, should work for VS2013, too. The usual trouble with handcrafted settings is that you cannot edit them in the project settings dialog, but that's ok IMO.

**Comment by rainers** [comment]
Thanks. 

Just had the idea to put the debug settings into the property sheet. It's simpler and more widely applied and can be removed from the project files. I can do that as a followup.

**Comment by rainers** [comment]
> I can do that as a followup.

Done: https://github.com/rainers/mago/commit/70893504e18d30714c6df05cdc17837437f07a74

**Pull request merged**